### ISSUE TITLE
Add additional test cases

### ISSUE THREAD

**Issue by CyrusSUEN** [dhowe/RiTa#128]
splitSentences() -- test this (use same tests as Java) and ask me if there are significant failures

minEditDist() -- make sure this has tests (also make sure its docs look correct)

_loadStringNode -- make sure this is getting tested for Node [called from RiTa.loadString(), RiTa.loadStrings(), RiMarkov.loadFrom(), RiGrammar.loadFrom() ]

timer() (ignore, removed from API)
stopTimer() (ignore, removed from API)
pauseTimer() (ignore, removed from API)

**Comment by CyrusSUEN** [comment]
> minEditDist() -- make sure this has tests (also make sure its docs look correct)

Already been added to tests.
But I can't find any JSON Doc about it.
Only javadocs exist before its method name in both RiTa and RiTaJS.

**Comment by CyrusSUEN** [comment]
> _loadStringNode -- make sure this is getting tested for Node [called from RiTa.loadString(), RiTa.loadStrings(), RiMarkov.loadFrom(), RiGrammar.loadFrom() ]

Yes, they are tested.
RiMarkov.loadFrom(), RiGrammar.loadFrom() and RiTa.loadStrings() will call
RiTa.loadString() which checks isNode() and calls _loadStringNode() during node tests.

> splitSentences() -- test this (use same tests as Java) and ask me if there are significant failures

Tested and no failure from both RiTa and RiTaJS.

**Comment by dhowe** [comment]
MinEditDist is an internal part of the object -- not part of the API, so no need for JSON docs. 
Is there more work required with RiTa.splitSentences() or can this be closed?

**Comment by CyrusSUEN** [comment]
> Is there more work required with RiTa.splitSentences()

No unless we need to rewrite its test case.

**Comment by dhowe** [comment]
If it does not have a full set of tests, then yes indeed, we have to write its test cases. Please check.

**Comment by CyrusSUEN** [comment]
Update test case in RiTa: https://github.com/dhowe/RiTa/pull/137

For RiTaJS, from src/rita.js:
```js
    splitSentences: function(text, regex) {

      var arr = text.match(/(\S.+?[.!?])(?=\s+|$)/g);

      return (text.length && arr && arr.length) ? arr : [text];
    },
```
Doesn't look like it can make use of the regex parameter.

And in the RiTaJS RiTa.splitSentences("") returns String[] with 1 length whereas in RiTa it returns the object with 0 length.

**Comment by CyrusSUEN** [comment]
Also found a TODO inside the test case: check Penn-Treebank splitting rules
I've briefly checked the rule but what is it for?

**Comment by dhowe** [comment]
These are the rules about how sentences should be split. Please create some additional tests (for Java and JS) based on these rules.

**Comment by CyrusSUEN** [comment]
Updated testSplitSentences() for RiTa: https://github.com/dhowe/RiTa/pull/177

commented out 2 failed tests
```java
    //  Q. or A. at start of sentence 
    input = "Q. Do I need a visa to visit Hong Kong? A. Visitors from most countries can enter Hong Kong without a visa for periods of seven to 180 days, depending on nationality.";
    output = RiTa.splitSentences(input);
    System.out.println(Arrays.asList(output));
    expected = new String[] { 
	"Q. Do I need a visa to visit Hong Kong?", 
	"A. Visitors from most countries can enter Hong Kong without a visa for periods of seven to 180 days, depending on nationality."};
    deepEqual(output, expected);
```
result: ```[Q. Do I need a visa to visit Hong Kong?, Visitors from most countries can enter Hong Kong without a visa for periods of seven to 180 days, depending on nationality.]```

```java
    // nextToken does not begin with an upper case
    input = "What did he buy? iPad or iPhone?";
    output = RiTa.splitSentences(input);
    expected = new String[] { 
	"What did he buy?",
	"iPad or iPhone?"};
    System.out.println(Arrays.asList(output));
    deepEqual(output, expected);
```
result: ```[What did he buy? iPad or iPhone?]```

Updated testSplitSentences() for RiTaJS: https://github.com/dhowe/RiTaJS/pull/60

**Comment by dhowe** [comment]
If you comment out a failing test that is valid, you NEED to add it to the KnownIssues test

**Comment by dhowe** [comment]
still waiting on confirmation @CyrusSUEN 
?

**Comment by CyrusSUEN** [comment]
> If you comment out a failing test that is valid, you NEED to add it to the KnownIssues test

Yes. This was done.
https://github.com/dhowe/RiTa/pull/177
https://github.com/CyrusSUEN/RiTa/commit/babb0127ddac6372c3fd9e54bb23143076068a67
